core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTArray.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt.part;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.CheckedInputStream;
+
+import org.eclipse.rdf4j.rio.hdt.util.CRC8;
+import org.eclipse.rdf4j.rio.hdt.util.VByte;
+
+/**
+ * HDT Array
+ * 
+ * This part starts with a byte indicating the type of the array, followed by a byte containing the number of bits used
+ * to encode an entry in the array, and the VByte-encoded number of entries.
+ *
+ * Then the 8-bit CRC, followed by the array data itself.
+ * 
+ * Structure:
+ * 
+ * <pre>
+ * +------+--------+---------+------+------...
+ * | type | nrbits | entries | CRC8 | data 
+ * +------+--------+---------+------+------...
+ * </pre>
+ * 
+ * @author Bart Hanssens
+ */
+public abstract class HDTArray extends HDTPart {
```

Comment:
Instead of sticking these components in a separate package and making them public, can we just stick them in the `rio.hdt` package and make them package-private instead? If I understand correctly, they're meant for internal use only, so I'd rather not have them be part of the public API if we can avoid it.

Reply:
Alternatively, we leave as-is and place our bets on eventually being able to upgrade to java 9, using modules to handle this kind of thing.

Reply:
Sure, I can move them to rio.hdt.
This is actually a good example where JDK 9 modules are useful :-)

They *might* need to become visible again, if at some point there would be a HDT Sail (since the format is actually optimized for handling simple ?s ?p ?o queries), but until then it would indeed be better to make these classes private.

Reply:
moved all classes to HDT package and made them package-private